boxes.cc

```diff
@@ -40,17 +44,42 @@ void BoxesTest::Boxes(const std::string &_physicsEngine, double _dt,
   ASSERT_GT(_modelCount, 0);
   std::string world_erb_path =
       boost::str(boost::format("%1%/boxes.world.erb") % WORLDS_DIR_PATH);
+  
 
   std::string world_path = boost::str(
       boost::format(
           "%1%/%2%/"
           "boxes_collision%3%_complex%4%_dt%5$.0e_modelCount%6%.world") %
       WORLDS_DIR_PATH % TEST_NAME % _collision % _complex % _dt % _modelCount);
 
+  std::string result_name = boost::str(
+    boost::format("%1%/%2%/MCAP/boxes_collision"
+            "%3%_complex%4%_dt%5$.0e_modelCount%6%_%7%.mcap") % RESULT_DIR_PATH %
+      TEST_NAME % _collision % _complex % _dt % _modelCount % _physicsEngine);
```

Comment:
It's a little confusing to me to have the `mcap` initialization code mixed with the world initialization code. I would prefer to group the related code together, for example as follows:

<details>

patch:

~~~
diff --git a/boxes.cc b/boxes.cc
index 417e7a0..4927460 100644
--- a/boxes.cc
+++ b/boxes.cc
@@ -42,25 +42,13 @@ using namespace benchmark;
 void BoxesTest::Boxes(const std::string &_physicsEngine, double _dt,
                       int _modelCount, bool _collision, bool _complex) {
   ASSERT_GT(_modelCount, 0);
-  std::string world_erb_path =
-      boost::str(boost::format("%1%/boxes.world.erb") % WORLDS_DIR_PATH);
-  
-
-  std::string world_path = boost::str(
-      boost::format(
-          "%1%/%2%/"
-          "boxes_collision%3%_complex%4%_dt%5$.0e_modelCount%6%.world") %
-      WORLDS_DIR_PATH % TEST_NAME % _collision % _complex % _dt % _modelCount);
 
+  // Initialize MCAP logging
   std::string result_name = boost::str(
     boost::format("%1%/%2%/MCAP/boxes_collision"
             "%3%_complex%4%_dt%5$.0e_modelCount%6%_%7%.mcap") % RESULT_DIR_PATH %
       TEST_NAME % _collision % _complex % _dt % _modelCount % _physicsEngine);
 
-  std::string command = boost::str(
-      boost::format(
-          "erb collision=%1% complex=%2% dt=%3% modelCount=%4% %5% > %6%") %
-      _collision % _complex % _dt % _modelCount % world_erb_path % world_path);
   // for logging all the model in boxes_model_count test case 
   bool log_all = false;
   // mcap writer 
@@ -81,12 +69,27 @@ void BoxesTest::Boxes(const std::string &_physicsEngine, double _dt,
     channelId = channel.id;
   }
 
+  // Generate world file
+  std::string world_erb_path =
+      boost::str(boost::format("%1%/boxes.world.erb") % WORLDS_DIR_PATH);
+
+  std::string world_path = boost::str(
+      boost::format(
+          "%1%/%2%/"
+          "boxes_collision%3%_complex%4%_dt%5$.0e_modelCount%6%.world") %
+      WORLDS_DIR_PATH % TEST_NAME % _collision % _complex % _dt % _modelCount);
+
+  std::string command = boost::str(
+      boost::format(
+          "erb collision=%1% complex=%2% dt=%3% modelCount=%4% %5% > %6%") %
+      _collision % _complex % _dt % _modelCount % world_erb_path % world_path);
+
   // creating model with desired configuration
   auto model_check = system(command.c_str());
   // checking if model is created
   ASSERT_EQ(model_check, 0);
 
-  // Load a blank world (no ground plane)
+  // Load the generated world file
   Load(world_path, true, _physicsEngine);
   physics::WorldPtr world = physics::get_world("default");
   ASSERT_NE(world, nullptr);
~~~

what the reorganized code could look like:

~~~
void BoxesTest::Boxes(const std::string &_physicsEngine, double _dt,
                      int _modelCount, bool _collision, bool _complex) {
  ASSERT_GT(_modelCount, 0);

  // Initialize MCAP logging
  std::string result_name = boost::str(
    boost::format("%1%/%2%/MCAP/boxes_collision"
            "%3%_complex%4%_dt%5$.0e_modelCount%6%_%7%.mcap") % RESULT_DIR_PATH %
      TEST_NAME % _collision % _complex % _dt % _modelCount % _physicsEngine);

  // for logging all the model in boxes_model_count test case 
  bool log_all = false;
  // mcap writer 
  mcap::McapWriter writer;
  auto options = mcap::McapWriterOptions("");
  const auto s = writer.open(result_name.c_str(), options);
  ASSERT_EQ(s.ok(), true);

  mcap::ChannelId channelId;

  {
    mcap::Schema schema(
      "benchmark_proto.Boxes_msg", "protobuf",
      foxglove::BuildFileDescriptorSet(benchmark_proto::Boxes_msg::descriptor()).SerializeAsString());
    writer.addSchema(schema);
    mcap::Channel channel("boxes_states", "protobuf", schema.id);
    writer.addChannel(channel);
    channelId = channel.id;
  }

  // Generate world file
  std::string world_erb_path =
      boost::str(boost::format("%1%/boxes.world.erb") % WORLDS_DIR_PATH);

  std::string world_path = boost::str(
      boost::format(
          "%1%/%2%/"
          "boxes_collision%3%_complex%4%_dt%5$.0e_modelCount%6%.world") %
      WORLDS_DIR_PATH % TEST_NAME % _collision % _complex % _dt % _modelCount);

  std::string command = boost::str(
      boost::format(
          "erb collision=%1% complex=%2% dt=%3% modelCount=%4% %5% > %6%") %
      _collision % _complex % _dt % _modelCount % world_erb_path % world_path);

  // creating model with desired configuration
  auto model_check = system(command.c_str());
  // checking if model is created
  ASSERT_EQ(model_check, 0);

  // Load the generated world file
  Load(world_path, true, _physicsEngine);
  physics::WorldPtr world = physics::get_world("default");
  ASSERT_NE(world, nullptr);


~~~

</details>

Reply:
yeah, this makes more sense